src/System.Text.Encodings.Web/src/System/Text/Unicode/UnicodeHelpers.cs

```diff
@@ -25,64 +25,43 @@ internal static unsafe class UnicodeHelpers
         /// </summary>
         internal const int UNICODE_LAST_CODEPOINT = 0x10FFFF;
 
-        private static uint[] _definedCharacterBitmap;
+        private static uint[] _definedCharacterBitmapMachineEndian;
 
         /// <summary>
-        /// Helper method which creates a bitmap of all characters which are
-        /// defined per the Unicode specification.
+        /// Returns a bitmap of all characters which are defined per the checked-in version
+        /// of the Unicode specification.
         /// </summary>
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private static uint[] CreateDefinedCharacterBitmap()
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static ReadOnlySpan<uint> GetDefinedCharacterBitmap()
         {
-            // The stream should be exactly 8KB in size.
-            var stream = typeof(UnicodeRange).GetTypeInfo().Assembly.GetManifestResourceStream("System.Text.Encodings.Web.Resources.unicode8definedcharacters.bin");
-
-            if (stream == null)
+            if (BitConverter.IsLittleEndian)
             {
-                throw new BadImageFormatException();
-            }
+                // Underlying data is a series of 32-bit little-endian values and is guaranteed
+                // properly aligned by the compiler, so we know this is a valid cast byte -> uint.
 
-            if (stream.Length != 8 * 1024)
-            {
-                Environment.FailFast("Corrupt data detected.");
+                return MemoryMarshal.Cast<byte, uint>(DefinedCharsBitmapSpan);
             }
-
-            // Read everything in as raw bytes.
-            byte[] rawData = new byte[8 * 1024];
-            for (int numBytesReadTotal = 0; numBytesReadTotal < rawData.Length;)
+            else
             {
-                int numBytesReadThisIteration = stream.Read(rawData, numBytesReadTotal, rawData.Length - numBytesReadTotal);
-                if (numBytesReadThisIteration == 0)
+                static uint[] CreateDefinedCharacterBitmapMachineEndian()
```

Comment:
>s [](start = 16, length = 1)

nit: why we are doing this as a method and not just a flat code. and maybe using InterlockExchange would used instead of Volatile read and write?

Reply:
Perhaps it'd be easier to have the field statically initialized rather than try to do it on first access? It'd refactor this code a little bit and probably make it more understandable. I'll experiment and see where it goes.